src/Promitor.Integrations.AzureMonitor/RequestHandlers/AzureResourceManagerThrottlingRequestHandler.cs

```diff
@@ -46,11 +54,19 @@ public AzureResourceManagerThrottlingRequestHandler(string tenantId, string subs
             _metricSinkWriter = metricSinkWriter;
             _metricsCollector = metricsCollector;
 
+            string id = applicationId;
+
+            if (authenticationMode == AuthenticationMode.ManagedIdentity)
+            {
+                id = string.IsNullOrEmpty(managedIdentityId) ? "system-assigned-identity" : managedIdentityId;
+            }
```

Comment:
Let's move this to a method
```suggestion
            DetermineApplicationId(...)
```